pkg/client/presentproof/client.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package presentproof
+
+import (
+	"errors"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/presentproof"
+)
+
+var (
+	errEmptyRequestPresentation = errors.New("request presentation message is empty")
+	errEmptyProposePresentation = errors.New("propose presentation message is empty")
+)
+
+// Provider contains dependencies for the protocol and is typically created by using aries.Context()
+type Provider interface {
+	Service(id string) (interface{}, error)
+}
+
+// ProtocolService defines the presentproof service.
+type ProtocolService interface {
+	service.DIDComm
+	Actions() ([]presentproof.Action, error)
+	ActionContinue(piID string, opt presentproof.Opt) error
+	ActionStop(piID string, err error) error
+}
+
+// Client enable access to presentproof API
+type Client struct {
+	service.Event
+	service ProtocolService
+}
+
+// New returns new instance of the presentproof client
+func New(ctx Provider) (*Client, error) {
+	raw, err := ctx.Service(presentproof.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	svc, ok := raw.(ProtocolService)
+	if !ok {
+		return nil, errors.New("cast service to presentproof service failed")
+	}
+
+	return &Client{
+		Event:   svc,
+		service: svc,
+	}, nil
+}
+
+// Actions returns unfinished actions for the async usage
+func (c *Client) Actions() ([]presentproof.Action, error) {
+	return c.service.Actions()
+}
+
+// SendRequestPresentation is used by the Verifier to send a request presentation.
+func (c *Client) SendRequestPresentation(msg *presentproof.RequestPresentation, myDID, theirDID string) error {
```

Comment:
I have no problems with this, but I notice that this is directly referencing models in the service package, which is a big no-no for reasons unclear to me. Maybe others can better explain why.

Reply:
Here is my answer. https://github.com/hyperledger/aries-framework-go/pull/1480#discussion_r395142761

Doing something like this is not a solution.
```
type RequestPresentation struct {
	*presentproof.RequestPresentation
}

```

Reply:
Ok, actually there is one way.

```
type RequestPresentation presentproof.RequestPresentation
// and then we can easily convert it
r1 := presentproof.RequestPresentation(*r)
fmt.Println(&r1)
```

Reply:
@llorllale  Done!

Reply:
@soluchok that does not address the point @rolsonquadras was (unsuccessfully) attempting to make in https://github.com/hyperledger/aries-framework-go/pull/1480#discussion_r395124308. These are only half-measures: true decoupling of the client and service packages would require more.

But like I said in my original comment above: I didn't have a problem with your original design anyway.